### PR TITLE
 Add gitlab projects subcommand (list by tag)

### PR DESCRIPTION
**Pull request by bittner**
Adds another sub-command for `gitlab` to list all projects matching one or more specific tags.

### CODE REVIEW

File: concierge_cli/manager.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    def __init__(self, uri=None, token=None):
+        """
+        Connects to a GitLab instance using connection details from one of the
+        local configuration files (see https://python-gitlab.readthedocs.io >
+        Configuration > Files). Connects to GitLab.com by default if no config
+        file is found. Specify an URI to override the config file lookup, the
+        token is optional (anonymous access if none is supplied).
+        """
+        if not uri:
```
Comment (andrekeller) [inline concierge_cli/manager.py]: shouldn't None be compared using is not None?

File: concierge_cli/manager.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            try:
+                self.api = Gitlab.from_config()
+            except GitlabConfigMissingError:
+                uri = DEFAULT_GITLAB_URI
+
+        if uri:
```
Comment (andrekeller) [inline concierge_cli/manager.py]: the if is redundant

File: concierge_cli/manager.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        """
+        List all projects and their topics, filtered by an optional
+        search pattern.
+        """
+        for group in self.api.groups.list(search=self.group_filter,
+                                          per_page=MAX_GROUPS):
+            for group_project in \
+                    group.projects.list(search=self.project_filter):
+                project = Project(self.api, group_project)
+                matched_tags = set(project.topic_list) & set(self.topic_list)
```
Comment (andrekeller) [inline concierge_cli/manager.py]: you could use set().intersection() which may be more obvious

File: setup.py [hunk 4]
```diff
@@ -1,19 +1,49 @@
...
-from setuptools import setup
+from os.path import abspath, dirname, join
+from setuptools import find_packages, setup
+
+import concierge_cli as package
+
+
+def read_file(filename):
+    """Get the contents of a file"""
+    here = abspath(dirname(__file__))
```
Comment (andrekeller) [inline setup.py]: why use a variable for this? it is not used anywhere else

File: concierge_cli/manager.py (hunk 2 above)
Comment (bittner) [reply concierge_cli/manager.py]: The `if` is relevant when the `GitlabConfigMissingError` is thrown.

File: concierge_cli/manager.py (hunk 3 above)
Comment (bittner) [reply concierge_cli/manager.py]: Sure, that's more explicit. I've tried it but it's not convincing me, it's just not that readable. :worried: 

With the `&` parameter I have two sets on both sides that I intersect. I keep it that way if you don't mind.

File: setup.py (hunk 4 above)
Comment (bittner) [reply setup.py]: I'll have @basharSaid fix this in the packaging task. :+1:

File: concierge_cli/manager.py (hunk 1 above)
Comment (bittner) [reply concierge_cli/manager.py]: An empty URI is just as useless here as `None`. I think the pythonic simplicity is acceptable here. Fine?

**Comment by bittner** [comment]
Thanks for reviewing, @andrekeller! :+1:

**Pull request merged**